triton-profiler/src/triton_profiler.rs

```diff
@@ -264,13 +266,16 @@ impl TritonProfiler {
         }
     }
 
-    pub fn finish(&mut self) {
+    pub fn finish(&mut self, cycle_count: Option<usize>) {
         assert!(!self.profile.is_empty(), "Nothing to finish.");
         assert!(
             self.stack.is_empty(),
             "Cannot finish before stack is empty."
         );
         self.total_time = self.timer.elapsed();
+        if let Some(cycle_count) = cycle_count {
+            self.clock_frequency = cycle_count / self.total_time.as_secs() as usize;
```

Comment:
Be aware of division-by-zero errors here. This is also the reason the test suite failed.